### PR TITLE
Single point of parsing for Time

### PR DESCRIPTION
**Pull request by alanconway**
All time parsing calls types.ParseTime()

Return errors from time parsing

Added types.ConvertError error type for all conversion errors.

Improved error messages and tightened up some error handling.

### CODE REVIEW

File: pkg/cloudevents/types/timestamp.go [hunk 1]
```diff
@@ -67,17 +60,10 @@ func (t *Timestamp) UnmarshalXML(d *xml.Decoder, start xml.StartElement) error {
...
 
 // String outputs the time using layout RFC3339.
-func (t *Timestamp) String() string {
-	if t == nil {
-		return time.Time{}.UTC().Format(time.RFC3339Nano)
-	}
-
-	return t.UTC().Format(time.RFC3339Nano)
-}
+func (t Timestamp) String() string { return t.UTC().Format(time.RFC3339Nano) }
```
Comment (n3wscott) [inline pkg/cloudevents/types/timestamp.go]: This should be the canonical form of the time. ToString(t) right?

Comment (alanconway) [reply pkg/cloudevents/types/timestamp.go]: Good point.

Comment (alanconway) [reply pkg/cloudevents/types/timestamp.go]: Fixed, removed all other refs to RFC399 outside value.go, also fixed usage from other packages.

**Comment by n3wscott** [comment]
LGTM!!

Looks like the commit has to be signed?

**Comment by alanconway** [comment]
Dammit, still haven't figured out how to make emacs do that for me. Should
be good now.

**Pull request merged**